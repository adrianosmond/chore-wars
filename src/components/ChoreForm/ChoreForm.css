.chore-form {
  display: flex;
  min-height: calc(100vh - 2rem);
  justify-content: center;
  align-items: center;
}

.form {
  width: 100%;
  max-width: 600px;
}

.form__option {
  display: flex;
  align-items: center;
  min-height: 2.5rem;
  margin-top: 0.5rem;
  cursor: pointer;
}

.form__row {
  display: flex;
  align-items: baseline;
}

.form__row-item:not(:first-child) {
  margin-left: 0.5rem;
}

.form__button-holder {
  display: grid;
  grid-template-columns: minmax(100px, 1fr)  minmax(100px, 1fr);
  grid-column-gap: 1rem;
  margin-top: 1rem;
}

.form__button-holder--tight {
  margin-top: 0;
}

@media screen and (min-width: 37.5em) {
  .form {
    padding: 1rem;
    max-width: 600px;
    margin-left: auto;
    margin: auto;
  }

  .form--contained {
    padding: 0;
  }
}
