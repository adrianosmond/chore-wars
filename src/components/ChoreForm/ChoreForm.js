import React from 'react';
import Card from 'components/Card';
import Button from 'components/Button';
import Typography from 'components/Typography';
import classes from './ChoreForm.module.css';

const ChoreForm = ({
  onSubmit,
  onCancel,
  prevQuestion,
  nextQuestion,
  children,
  hasError,
  isBusy,
  title,
}) => {
  return (
    <form onSubmit={onSubmit} className={classes.form}>
      <Typography appearance="h1">{title}</Typography>
      <Card appearance="medium">
        <div className={classes.question}>{children}</div>
        <div className={classes.buttons}>
          {prevQuestion && (
            <Button onClick={prevQuestion} appearance="secondary">
              Previous
            </Button>
          )}

          {!prevQuestion && (
            <Button onClick={onCancel} appearance="secondary">
              Cancel
            </Button>
          )}

          {nextQuestion && (
            <Button onClick={nextQuestion} disabled={hasError}>
              Next
            </Button>
          )}

          {!nextQuestion && (
            <Button type="submit" disabled={hasError} isBusy={isBusy}>
              Done
            </Button>
          )}
        </div>
      </Card>
    </form>
  );
};

export default ChoreForm;
