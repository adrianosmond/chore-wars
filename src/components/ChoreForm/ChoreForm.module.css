.form {
  max-width: 32rem;
  margin: var(--spacing-s) auto 0;
}

.question {
  margin-bottom: var(--spacing-l);
}

.buttons {
  display: flex;
  justify-content: space-between;
}

.buttons > :only-child {
  margin-left: auto;
}
