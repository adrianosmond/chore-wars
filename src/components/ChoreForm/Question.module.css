.radioLabelWrapper {
  display: flex;
  align-items: center;
}

.radioLabelWrapper > * {
  margin-right: var(--spacing-xs);
}
