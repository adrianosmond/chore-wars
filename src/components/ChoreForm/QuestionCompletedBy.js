import React, { useMemo } from 'react';
import Select from 'react-select';
import { usePlayers } from 'contexts/game';
import ChoreQuestion from 'components/ChoreQuestion';

const QuestionCompletedBy = ({ currentPlayer, onChange }) => {
  const players = usePlayers();

  const playerOptions = useMemo(
    () =>
      players.map(player => ({
        value: player.id,
        label: player.name,
      })),
    [players],
  );

  const initialValue = playerOptions.find(
    player => player.value === currentPlayer,
  );

  return (
    <ChoreQuestion question="Who forgot to log the chore?">
      <Select
        options={playerOptions}
        defaultValue={initialValue}
        onChange={onChange}
      ></Select>
    </ChoreQuestion>
  );
};

export default QuestionCompletedBy;
