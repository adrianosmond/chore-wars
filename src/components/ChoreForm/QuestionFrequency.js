import React, { useState, useCallback, useEffect } from 'react';
import { useChoreForm } from 'contexts/choreForm';
import { MIN_CHORE_FREQUENCY, MAX_CHORE_FREQUENCY } from 'constants/constants';

import ChoreQuestion from 'components/ChoreQuestion';
import Input from 'components/Input';
import Radio from 'components/Radio';

import classes from './Question.module.css';

const isFrequencyInvalid = frequency =>
  Number.isNaN(frequency) ||
  typeof frequency !== 'number' ||
  frequency < MIN_CHORE_FREQUENCY ||
  frequency > MAX_CHORE_FREQUENCY;

const QuestionFrequency = () => {
  const { frequency, updateFrequency, setHasError } = useChoreForm();
  const [hasFrequency, setHasFrequency] = useState(frequency !== 0);
  const frequencyOn = useCallback(() => {
    setHasFrequency(true);
  }, []);
  const frequencyOff = useCallback(() => {
    setHasFrequency(false);
  }, []);

  useEffect(() => {
    const freq = parseInt(frequency, 10);
    setHasError(hasFrequency && isFrequencyInvalid(freq));
  }, [frequency, hasFrequency, setHasError]);

  return (
    <ChoreQuestion question="How often do you need to do this chore?">
      <Radio
        name="frequency-type"
        value="with-regularity"
        checked={hasFrequency}
        onChange={frequencyOn}
        label={
          <div className={classes.radioLabelWrapper}>
            <span>Every </span>
            <Input
              type="number"
              pattern="[0-9]*"
              min={MIN_CHORE_FREQUENCY}
              max={MAX_CHORE_FREQUENCY}
              maxLength={MAX_CHORE_FREQUENCY.toString().length}
              value={frequency}
              onChange={updateFrequency}
              onClick={frequencyOn}
            />
            <span>days</span>
          </div>
        }
      />
      <Radio
        name="frequency-type"
        value="as-and-when"
        checked={!hasFrequency}
        onChange={frequencyOff}
        label="Whenever it needs doing"
      />
    </ChoreQuestion>
  );
};

export default QuestionFrequency;
