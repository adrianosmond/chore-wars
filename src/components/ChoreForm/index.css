.chore-form {
  display: flex;
  min-height: calc(100vh - 2rem);
  justify-content: center;
  align-items: center;
}

.form {
  width: 100%;
  max-width: 600px;
}

.form__label {
  display: block;
  margin-top: 1rem;
  margin-bottom: 2rem;
  font-family: 'Patua One', sans-serif;
  /* font-weight: 400; */
  font-size: 1.5em;
}

.form__option {
  display: flex;
  align-items: center;
  min-height: 2.5rem;
  margin-top: 0.5rem;
  cursor: pointer;
}

.form__label--error {
  color: crimson;
}

.form__label:first-child {
  margin-top: 0;
}

.form__row {
  display: flex;
  align-items: baseline;
}

.form__row--inactive {
  opacity: 0.8;
}

.form__row-item:not(:first-child) {
  margin-left: 0.5rem;
}

.form__input {
  display: block;
  padding: 0.5rem;
  margin-top: 0.25rem;
  width: 100%;
  border: 1px solid #ccc;
  font: inherit;
}

.form__input[maxlength="3"] {
  width: 70px;
}

.form__input[id="doneDate"] {
  -webkit-appearance: none;
}

.form__button-holder {
  display: grid;
  grid-template-columns: minmax(100px, 1fr)  minmax(100px, 1fr);
  grid-column-gap: 1rem;
  margin-top: 1rem;
}

.form__button-holder--tight {
  margin-top: 0;
}

.form__button {
  display: block;
  margin: 0;
  margin-top: 1rem;
  padding: 1rem;
  border: none;
  border-bottom: 2px solid rgba(0,0,0,0.3);
  width: 100%;
  background-color: darkcyan;
  color: #fff;
  font: inherit;
  text-align: center;
  text-decoration: none;
  line-height: 1;
  border-radius: 0.25rem;
  outline-color: #ccc;
  cursor: pointer;
}

.form__button:active {
  border-bottom: none;
  border-top: 2px solid transparent;
}

.form__button:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.form__button--secondary {
  background-color: darkslateblue;
}

.form__button--tertiary {
  background-color: dodgerblue;
}

.form__button--quaternary {
  background-color:#fff;
  color: darkslateblue;
}

@media screen and (min-width: 37.5em) {
  .form {
    padding: 1rem;
    max-width: 600px;
    margin-left: auto;
    margin: auto;
  }

  .form--contained {
    padding: 0;
  }
}
