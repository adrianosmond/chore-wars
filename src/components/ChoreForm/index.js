import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import fecha from 'fecha';

import * as routes from '../../constants/routes';
import { makeSlug } from '../../constants/utils';
import { MIN_CHORE_FREQUENCY, MAX_CHORE_FREQUENCY, MIN_CHORE_POINTS, MAX_CHORE_POINTS } from '../../constants/constants';

import './index.css';

const FormQuestion = (props) => {
  if (props.id !== props.currentQuestionId) return null;
  const TagName = props.noLabel ? 'div' : 'label';
  return (
    <div className="form__question">
      <TagName>
        <span className={`form__label${props.error ? ' form__label--error' : ''}`}>
          {props.label}
        </span>
        {props.children}
      </TagName>
    </div>
  );
};

const defaultQuestions = ['title', 'frequency', 'pointsPerTime', 'lastDone'];

class ChoreForm extends Component {
  constructor(props) {
    super(props);

    const { chore } = props;
    const questions = props.questions || defaultQuestions;
    this.state = {
      ...chore,
      slug: makeSlug(chore.title),
      doneDate: fecha.format(chore.lastDone, 'YYYY-MM-DD'),
      choreFrequency: chore.frequency === 0 ? 'as-and-when' : 'with-regularity',
      currentQuestionId: questions[0],
      questions,
      currentTime: props.currentTime,
      error: false,
    };
  }

  formIsValid() {
    if (this.state.currentQuestionId === 'title') {
      if (this.state.title.trim().length === 0) {
        this.setState({
          error: true,
        });
        return false;
      }
    }
    if (this.state.currentQuestionId === 'frequency') {
      if (this.state.choreFrequency === 'with-regularity') {
        const freq = parseInt(this.state.frequency, 10);
        if (typeof freq !== 'number' || freq < MIN_CHORE_FREQUENCY || freq > MAX_CHORE_FREQUENCY) {
          this.setState({
            error: true,
          });
          return false;
        }
      }
    }
    if (this.state.currentQuestionId === 'pointsPerTime') {
      const points = parseInt(this.state.pointsPerTime, 10);
      if (typeof points !== 'number' || points < 0 || points > 200) {
        this.setState({
          error: true,
        });
        return false;
      }
    }
    if (this.state.currentQuestionId === 'doneDate') {
      const date = parseInt(this.state.lastDone, 10);
      if (typeof date !== 'number' || date < 0 || date > this.props.currentTime) {
        this.setState({
          error: true,
        });
        return false;
      }
    }
    if (this.state.currentQuestionId === 'forgotToLog') {
      const date = parseInt(this.state.currentTime, 10);
      if (typeof date !== 'number' || date < 0 || date > this.props.currentTime) {
        this.setState({
          error: true,
        });
        return false;
      }
    }
    return true;
  }

  isFirstQuestion() {
    const { currentQuestionId, questions } = this.state;
    return questions.indexOf(currentQuestionId) === 0;
  }

  isLastQuestion() {
    const { currentQuestionId, questions } = this.state;
    return questions.indexOf(currentQuestionId) === questions.length - 1;
  }

  nextQuestion(event) {
    event.preventDefault();
    if (!this.formIsValid()) return;
    const { currentQuestionId, questions } = this.state;
    const currentIndex = questions.indexOf(currentQuestionId);
    this.setState({
      error: false,
      currentQuestionId: questions[currentIndex + 1],
    });
  }

  prevQuestion(event) {
    event.preventDefault();
    const { currentQuestionId, questions } = this.state;
    const currentIndex = questions.indexOf(currentQuestionId);
    this.setState({
      error: false,
      currentQuestionId: questions[currentIndex - 1],
    });
  }

  onSubmit(event) {
    event.preventDefault();

    if (!this.formIsValid()) return false;

    const chore = {
      title: this.state.title,
      frequency: parseInt(this.state.frequency, 10),
      pointsPerTime: parseInt(this.state.pointsPerTime, 10),
      lastDone: this.state.lastDone,
    };

    this.props.onSubmit(chore, this.state.slug);
    return false;
  }

  render() {
    return (
      <div className="chore-form">
        <form onSubmit={this.onSubmit.bind(this)} className="form">
          <FormQuestion id="title" currentQuestionId={this.state.currentQuestionId}
            label="What is the name of this chore?" error={this.state.error}>
            <input className="form__input" id="title" type="text" value={this.state.title} onChange={(event) => { this.setState({ title: event.target.value, slug: makeSlug(event.target.value) }); }} />
          </FormQuestion>
          <FormQuestion id="frequency" currentQuestionId={this.state.currentQuestionId}
            label="How often do you need to do this chore?" error={this.state.error}
            noLabel={true}>
            <label className="form__option">
              <div className="form__row">
                <input type="radio" name="frequency-type" value="with-regularity"
                  checked={this.state.choreFrequency === 'with-regularity'}
                  onChange={() => this.setState({ choreFrequency: 'with-regularity' })} />
                <span className="form__row-item">Every </span>
                <input className="form__row-item form__input" id="frequency" type="number" pattern="[0-9]*"
                  min={MIN_CHORE_FREQUENCY} max={MAX_CHORE_FREQUENCY} maxLength={MAX_CHORE_FREQUENCY.toString().length}
                  value={this.state.frequency} onChange={(event) => { this.setState({ frequency: event.target.value }); }}
                  onClick={() => this.setState({ choreFrequency: 'with-regularity' })} />
                <span className="form__row-item">days</span>
              </div>
            </label>
            <label className="form__option">
            <div className="form__row form__row--inactive">
                <input type="radio" name="frequency-type" value="as-and-when"
                  checked={this.state.choreFrequency === 'as-and-when'}
                  onChange={() => this.setState({ choreFrequency: 'as-and-when' })} />
                <span className="form__row-item">Whenever it needs doing</span>
              </div>
            </label>
          </FormQuestion>
          <FormQuestion id="pointsPerTime" currentQuestionId={this.state.currentQuestionId}
            label="How many points should this chore be worth?" error={this.state.error}>
            <p></p>
            <input className="form__input" id="pointsPerTime" type="number" pattern="[0-9]*"
              min={MIN_CHORE_POINTS} max={MAX_CHORE_POINTS} maxLength={MAX_CHORE_POINTS.toString().length}
              value={this.state.pointsPerTime} onChange={(event) => { this.setState({ pointsPerTime: event.target.value }); }} />
          </FormQuestion>
          <FormQuestion id="lastDone" currentQuestionId={this.state.currentQuestionId}
            label="When was this chore last done?" error={this.state.error}>
            <input className="form__input" id="doneDate" type="date"
            value={this.state.doneDate} max={fecha.format(this.props.currentTime, 'YYYY-MM-DD')}
            onChange={(event) => { this.setState({ doneDate: event.target.value, lastDone: new Date(event.target.value).getTime() }); }} />
          </FormQuestion>
          <FormQuestion id="forgotToLog" currentQuestionId={this.state.currentQuestionId}
            label="When did you do this chore?" error={this.state.error}>
            <input className="form__input" id="doneDate" type="datetime-local"
              value={fecha.format(this.state.currentTime, 'YYYY-MM-DDTHH:mm')} max={fecha.format(this.props.currentTime, 'YYYY-MM-DD')}
              onChange={(event) => {
                const currentTime = new Date(event.target.value).getTime();
                if (currentTime) {
                  const offset = new Date().getTimezoneOffset() * 60000;
                  this.setState({
                    currentTime,
                    lastDone: currentTime - offset,
                  });
                }
              }} />
          </FormQuestion>

          <div className="form__button-holder">
            { this.isFirstQuestion() ? <Link className="form__button form__button--secondary" to={routes.CHORES}>Cancel</Link> : null }
            { !this.isFirstQuestion() ? <button className="form__button form__button--secondary" onClick={this.prevQuestion.bind(this)}>Back</button> : null }
            { !this.isLastQuestion() ? <button className="form__button" onClick={this.nextQuestion.bind(this)}>Continue</button> : null }
            { this.isLastQuestion() ? <button className="form__button" type="submit">Save Chore</button> : null }
          </div>
        </form>
      </div>
    );
  }
}

export default ChoreForm;
