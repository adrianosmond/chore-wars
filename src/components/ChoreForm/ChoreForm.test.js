import React from 'react';
import { shallow } from 'enzyme';
import ChoreForm from './index';

const now = new Date().getTime();
const choreToEdit = {
  title: 'Test Chore',
  slug: 'test-chore',
  frequency: 1,
  pointsPerTime: 100,
  lastDone: now,
};
const newChore = {
  title: '',
  slug: '',
  frequency: 7,
  pointsPerTime: 10,
  lastDone: now,
};
const submitHandler = jest.fn();

describe('Chore Form', () => {
  it('Renders a new chore', () => {
    expect(shallow(<ChoreForm chore={newChore} onSubmit={submitHandler} />)).toMatchSnapshot();
  });

  it('Renders an existing chore', () => {
    expect(shallow(<ChoreForm chore={choreToEdit} onSubmit={submitHandler} />)).toMatchSnapshot();
  });

  it('Handles button clicks', () => {
    const fakeEvent = { preventDefault: () => {} };
    const shallowForm = shallow(<ChoreForm chore={choreToEdit} onSubmit={submitHandler} />);
    shallowForm.find('form').simulate('submit', fakeEvent);
    expect(submitHandler).toHaveBeenCalled();
  });
});
