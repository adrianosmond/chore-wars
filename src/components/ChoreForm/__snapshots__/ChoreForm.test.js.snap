// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chore Form Renders a new chore 1`] = `
<div
  className="chore-form"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="title"
      label="What is the name of this chore?"
    >
      <input
        className="form__input"
        id="title"
        onChange={[Function]}
        type="text"
        value=""
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="frequency"
      label="How often do you need to do this chore?"
      noLabel={true}
    >
      <label
        className="form__option"
      >
        <div
          className="form__row"
        >
          <input
            checked={true}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="with-regularity"
          />
          <span
            className="form__row-item"
          >
            Every 
          </span>
          <input
            className="form__row-item form__input"
            id="frequency"
            max={365}
            maxLength={3}
            min={1}
            onChange={[Function]}
            onClick={[Function]}
            pattern="[0-9]*"
            type="number"
            value={7}
          />
          <span
            className="form__row-item"
          >
            days
          </span>
        </div>
      </label>
      <label
        className="form__option"
      >
        <div
          className="form__row form__row--inactive"
        >
          <input
            checked={false}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="as-and-when"
          />
          <span
            className="form__row-item"
          >
            Whenever it needs doing
          </span>
        </div>
      </label>
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="pointsPerTime"
      label="How many points should this chore be worth?"
    >
      <p />
      <input
        className="form__input"
        id="pointsPerTime"
        max={200}
        maxLength={3}
        min={1}
        onChange={[Function]}
        pattern="[0-9]*"
        type="number"
        value={10}
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="lastDone"
      label="When was this chore last done?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="date"
        value="2018-02-18"
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="forgotToLog"
      label="When did you do this chore?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="datetime-local"
        value="2018-02-18T01:00"
      />
    </FormQuestion>
    <div
      className="form__button-holder"
    >
      <Link
        className="form__button form__button--secondary"
        replace={false}
        to="/chores"
      >
        Cancel
      </Link>
      <button
        className="form__button"
        onClick={[Function]}
      >
        Continue
      </button>
    </div>
  </form>
</div>
`;

exports[`Chore Form Renders a specified question 1`] = `
<div
  className="chore-form"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormQuestion
      currentQuestionId="forgotToLog"
      error={false}
      id="title"
      label="What is the name of this chore?"
    >
      <input
        className="form__input"
        id="title"
        onChange={[Function]}
        type="text"
        value="Test Chore"
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="forgotToLog"
      error={false}
      id="frequency"
      label="How often do you need to do this chore?"
      noLabel={true}
    >
      <label
        className="form__option"
      >
        <div
          className="form__row"
        >
          <input
            checked={true}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="with-regularity"
          />
          <span
            className="form__row-item"
          >
            Every 
          </span>
          <input
            className="form__row-item form__input"
            id="frequency"
            max={365}
            maxLength={3}
            min={1}
            onChange={[Function]}
            onClick={[Function]}
            pattern="[0-9]*"
            type="number"
            value={1}
          />
          <span
            className="form__row-item"
          >
            days
          </span>
        </div>
      </label>
      <label
        className="form__option"
      >
        <div
          className="form__row form__row--inactive"
        >
          <input
            checked={false}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="as-and-when"
          />
          <span
            className="form__row-item"
          >
            Whenever it needs doing
          </span>
        </div>
      </label>
    </FormQuestion>
    <FormQuestion
      currentQuestionId="forgotToLog"
      error={false}
      id="pointsPerTime"
      label="How many points should this chore be worth?"
    >
      <p />
      <input
        className="form__input"
        id="pointsPerTime"
        max={200}
        maxLength={3}
        min={1}
        onChange={[Function]}
        pattern="[0-9]*"
        type="number"
        value={100}
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="forgotToLog"
      error={false}
      id="lastDone"
      label="When was this chore last done?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="date"
        value="2018-02-18"
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="forgotToLog"
      error={false}
      id="forgotToLog"
      label="When did you do this chore?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="datetime-local"
        value="2018-02-18T01:00"
      />
    </FormQuestion>
    <div
      className="form__button-holder"
    >
      <Link
        className="form__button form__button--secondary"
        replace={false}
        to="/chores"
      >
        Cancel
      </Link>
      <button
        className="form__button"
        type="submit"
      >
        Save Chore
      </button>
    </div>
  </form>
</div>
`;

exports[`Chore Form Renders an existing chore 1`] = `
<div
  className="chore-form"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="title"
      label="What is the name of this chore?"
    >
      <input
        className="form__input"
        id="title"
        onChange={[Function]}
        type="text"
        value="Test Chore"
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="frequency"
      label="How often do you need to do this chore?"
      noLabel={true}
    >
      <label
        className="form__option"
      >
        <div
          className="form__row"
        >
          <input
            checked={true}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="with-regularity"
          />
          <span
            className="form__row-item"
          >
            Every 
          </span>
          <input
            className="form__row-item form__input"
            id="frequency"
            max={365}
            maxLength={3}
            min={1}
            onChange={[Function]}
            onClick={[Function]}
            pattern="[0-9]*"
            type="number"
            value={1}
          />
          <span
            className="form__row-item"
          >
            days
          </span>
        </div>
      </label>
      <label
        className="form__option"
      >
        <div
          className="form__row form__row--inactive"
        >
          <input
            checked={false}
            name="frequency-type"
            onChange={[Function]}
            type="radio"
            value="as-and-when"
          />
          <span
            className="form__row-item"
          >
            Whenever it needs doing
          </span>
        </div>
      </label>
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="pointsPerTime"
      label="How many points should this chore be worth?"
    >
      <p />
      <input
        className="form__input"
        id="pointsPerTime"
        max={200}
        maxLength={3}
        min={1}
        onChange={[Function]}
        pattern="[0-9]*"
        type="number"
        value={100}
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="lastDone"
      label="When was this chore last done?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="date"
        value="2018-02-18"
      />
    </FormQuestion>
    <FormQuestion
      currentQuestionId="title"
      error={false}
      id="forgotToLog"
      label="When did you do this chore?"
    >
      <input
        className="form__input"
        id="doneDate"
        max="2018-02-18"
        onChange={[Function]}
        type="datetime-local"
        value="2018-02-18T01:00"
      />
    </FormQuestion>
    <div
      className="form__button-holder"
    >
      <Link
        className="form__button form__button--secondary"
        replace={false}
        to="/chores"
      >
        Cancel
      </Link>
      <button
        className="form__button"
        onClick={[Function]}
      >
        Continue
      </button>
    </div>
  </form>
</div>
`;
